pkg/server/projectconfig/prebuild.go

```diff
@@ -0,0 +1,316 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package projectconfig
+
+import (
+	"fmt"
+
+	"github.com/daytonaio/daytona/internal/util"
+	"github.com/daytonaio/daytona/pkg/build"
+	"github.com/daytonaio/daytona/pkg/gitprovider"
+	"github.com/daytonaio/daytona/pkg/server/projectconfig/dto"
+	"github.com/daytonaio/daytona/pkg/workspace/project/config"
+)
+
+func (s *ProjectConfigService) SetPrebuild(createPrebuildDto dto.CreatePrebuildDTO) (*dto.PrebuildDTO, error) {
+	projectConfig, err := s.Find(&config.Filter{
+		Name: &createPrebuildDto.ProjectConfigName,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	existingPrebuild, err := projectConfig.FindPrebuild(&config.PrebuildFilter{
+		Branch: &createPrebuildDto.Branch,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if existingPrebuild != nil {
+		return nil, fmt.Errorf("prebuild for the specified project config and branch already exists")
+	}
+
+	gitProvider, gitProviderId, err := s.gitProviderService.GetGitProviderForUrl(projectConfig.RepositoryUrl)
+	if err != nil {
+		return nil, err
+	}
+
+	repository, err := gitProvider.GetRepositoryFromUrl(projectConfig.RepositoryUrl)
+	if err != nil {
+		return nil, err
+	}
+
+	if projectConfig.WebhookId == nil {
+		webhookId, err := s.gitProviderService.RegisterPrebuildWebhook(gitProviderId, repository, s.prebuildWebhookEndpoint)
+		if err != nil {
+			return nil, err
+		}
+
+		projectConfig.WebhookId = &webhookId
+
+		err = s.configStore.Save(projectConfig)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	prebuild := &config.PrebuildConfig{
+		Branch:         createPrebuildDto.Branch,
+		CommitInterval: createPrebuildDto.CommitInterval,
+		TriggerFiles:   createPrebuildDto.TriggerFiles,
+	}
+
+	err = prebuild.GenerateId()
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: handle webhook registration removal if prebuild creation fails
+	err = projectConfig.SetPrebuild(prebuild)
+	if err != nil {
+		return nil, err
+	}
+
+	err = s.configStore.Save(projectConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	if createPrebuildDto.RunAtInit != nil && *createPrebuildDto.RunAtInit {
```

Comment:
I suggest we remove this and have this only be a client option.
E.g. `daytona pb add --run` will create a prebuild and then run a request that runs the build.